src/form-engine.component.tsx

```diff
@@ -56,35 +55,40 @@ const FormEngine = ({
   const sessionDate = useMemo(() => {
     return new Date();
   }, []);
-  const workspaceLayout = useWorkspaceLayout(ref);
+  const workspaceSize = useFormWorkspaceSize(ref);
   const { patient, isLoadingPatient } = usePatientData(patientUUID);
   const [isLoadingDependencies, setIsLoadingDependencies] = useState(false);
-  const [showSidebar, setShowSidebar] = useState(false);
   const [isSubmitting, setIsSubmitting] = useState(false);
   const [isFormDirty, setIsFormDirty] = useState(false);
   const sessionMode = !isEmpty(mode) ? mode : !isEmpty(encounterUUID) ? 'edit' : 'enter';
   const { isFormExpanded, hideFormCollapseToggle } = useFormCollapse(sessionMode);
+  const { hasMultiplePages } = usePageObserver();
 
-  // TODO: Updating this prop triggers a rerender of the entire form. This means whenever we scroll into a new page, the form is rerendered.
-  // Figure out a way to avoid this. Maybe use a ref with an observer instead of a state?
-  const [currentPage, setCurrentPage] = useState('');
   const {
     formJson: refinedFormJson,
     isLoading: isLoadingFormJson,
     formError,
   } = useFormJson(formUUID, formJson, encounterUUID, formSessionIntent);
 
   const showPatientBanner = useMemo(() => {
-    return patient && workspaceLayout !== 'minimized' && mode !== 'embedded-view';
-  }, [patient, mode, workspaceLayout]);
+    return patient && workspaceSize === 'ultra-wide' && mode !== 'embedded-view';
+  }, [patient, mode, workspaceSize]);
 
   const showButtonSet = useMemo(() => {
-    // if (mode === 'embedded-view') {
-    //   return false;
-    // }
-    // return workspaceLayout === 'minimized' || (workspaceLayout === 'maximized' && scrollablePages.size <= 1);
-    return true;
-  }, [mode, workspaceLayout]);
+    if (mode === 'embedded-view' || isLoadingDependencies || hasMultiplePages === null) {
+      return false;
+    }
+
+    return ['narrow', 'wider'].includes(workspaceSize) || !hasMultiplePages;
+  }, [mode, workspaceSize, isLoadingDependencies, hasMultiplePages]);
+
+  const showSidebar = useMemo(() => {
+    if (mode === 'embedded-view' || isLoadingDependencies || hasMultiplePages === null) {
+      return false;
+    }
+
+    return ['extra-wide', 'ultra-wide'].includes(workspaceSize) && hasMultiplePages;
```

Comment:
These modes are starting to get confusing. This implies 4 modes yet the design guidelines as well as workspace only has 3 modes. Does `extra-wide` have a different behavour from `ultra-wide`

Reply:
`ultra-wide` is basically "full screen". See the screenshots in the PR.

Reply:
Thanks @ibacher i saw that. Then i guess the issue would be with `extra-wide` unless that naming is being referenced else where within the patient chart.